src/Microsoft.TestPlatform.PlatformAbstractions/common/System/ProcessHelper.cs

```diff
@@ -131,15 +131,18 @@ public bool TryGetExitCode(object process, out int exitCode)
         }
 
         /// <inheritdoc/>
-        public void SetExitCallback(int processId, Action callbackAction)
+        public void SetExitCallback(int processId, Action<object> callbackAction)
         {
-            var process = Process.GetProcessById(processId);
-
-            process.EnableRaisingEvents = true;
-            process.Exited += (sender, args) =>
+            try
+            {
+                var process = Process.GetProcessById(processId);
+                process.EnableRaisingEvents = true;
+                process.Exited += (sender, args) => callbackAction.Invoke(process);
```

Comment:
>callbackAction.Invoke(process); [](start = 52, length = 31)

if there's a process crash before the hookup.. call the handler..

Reply:
Done.